app/components/UI/ApproveTransactionReview/index.js

```
@@ -653,7 +653,9 @@ class ApproveTransactionReview extends PureComponent {
   goToSpendCap = () => this.setState({ isReadyToApprove: false });
 
   customSpendInputValid = (value) => {
-    this.setState({ isCustomSpendInputValid: value });
+    if (this.state.token?.tokenStandard === ERC20) {
+      this.setState({ isCustomSpendInputValid: value });
+    }
```

wrapping this with the ERC20 check does not seem necessary since it seems the UI with access to this method is only shown when the token is an ERC20 token. 

If we are adding a check here, I think we should include a warning rather than having a case like this fail silently